src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -715,6 +715,25 @@ define('diagramStore', ['jquery', 'xwiki-meta', 'draw.io', 'xwiki-events-bridge'
     oldImportFiles.apply(this, newArgs);
   }
 
+  //
+  // Change the URL generated by export for containing the actual host.
+  //
+  var oldCreateLink = EditorUi.prototype.createLink;
+  EditorUi.prototype.createLink = function() {
+    let rawURL = oldCreateLink.apply(this, arguments);
+    // Do not include '#' because it's automatically added by getURL function below.
+    let documentFragment = rawURL.substring(rawURL.indexOf('#') + 1);
+    if (documentFragment.substring(0, 1) == 'R') {
+      let queryString = rawURL.substring (rawURL.indexOf('/?'), rawURL.indexOf('#R'));
```

Comment:
Why are you looking for ``/?``. The query string is separated by ``?``, which can be preceded by ``/`` but it's not mandatory. Also, what if the query string is not present? (e.g. draw.io might leave the query string empty if the user doesn't select any option from the export dialog)